netweaver/monitoring.sls

```diff
@@ -5,22 +5,30 @@ prometheus_sap_host_exporter_pkg:
   pkg.installed:
     - name: prometheus-sap_host_exporter
 
-sap_host_exporter_exporter_service:
+# the sid, instance number pair of a node is unique, so we need to adapt configuration
+{% for node in netweaver.nodes if host == node.host and node.sap_instance != "db" %}
+{% set instance = node.instance %}
+{% set instance = '{:0>2}'.format(node.instance) %}
+{% set daemon_instance = '{}_{}'.format(node.sid, instance) %}
+
+sap_host_exporter_exporter_service_{{ daemon_instance }}:
   service.running:
-    - name: prometheus-sap_host_exporter
+    - name: prometheus-sap_host_exporter@{{ daemon_instance }}
     - enable: True
     - restart: True
     - require:
       - pkg: prometheus_sap_host_exporter_pkg
-      - file: sap_host_exporter_configuration
+      - file: sap_host_exporter_configuration_{{ daemon_instance }}
+    - watch:
+      - file: sap_host_exporter_configuration_{{ daemon_instance }}
+
 
-# for each node the instance number changes so we need to adapt configuration
-{% for node in netweaver.nodes if host == node.host %}
-sap_host_exporter_configuration:
+# we can have multiple exporters in the same host in that moment, we need to change the port used by the exporter
+sap_host_exporter_configuration_{{ daemon_instance }}:
   file.managed:
-    - name: /etc/sap_host_exporter.yaml
+    - name: /etc/sap_host_exporter{{ daemon_instance }}.yaml
     - contents: |
-         port: "9680"
+         port: "968{{ loop.index0 }}"
          address: "0.0.0.0"
          log-level: "info"
          sap-control-url: "http://127.0.0.1:5{{ node.instance }}13"
```

Comment:
Don't use the `{{ node.instance }}` if the value is `0` it will create the port `5013` what is incorrect.
Use the `{{ instance }}` that was created before fixing this issue

Reply:
done thx